Classes/Event/EnrichScriptTagEvent.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the "matomo_integration" extension for TYPO3 CMS.
+ *
+ * For the full copyright and license information, please read the
+ * LICENSE.txt file that was distributed with this source code.
+ */
+
+namespace Brotkrueml\MatomoIntegration\Event;
+
+use Brotkrueml\MatomoIntegration\Exceptions\InvalidDataAttributeName;
+use Psr\Http\Message\ServerRequestInterface;
+
+/**
+ * This event allows the developer to enrich the build script tag with additional attributes.
+ */
+final class EnrichScriptTagEvent
+{
+    /**
+     * The server request in case the developer requires additional information.
+     */
+    private ServerRequestInterface $serverRequest;
+
+    /**
+     * The id of the element
+     */
+    private string $id = '';
+
+    /**
+     * The type that should be set for the script tag. I.e. text/plain
+     */
+    private string $type = '';
+
+    /**
+     * Additional data attributes.
+     * @var array <string, string|null>
+     */
+    private array $dataAttributes = [];
+
+    public function __construct(ServerRequestInterface $serverRequest)
+    {
+        $this->serverRequest = $serverRequest;
```

Comment:
`$this->request` should be fine, as also the core uses this for the server request (see e.g. https://github.com/TYPO3/typo3/blob/main/typo3/sysext/backend/Classes/Controller/LoginController.php#L120) - so this is a common usage.

Reply:
Changed.